frontend/fingertips-frontend/components/organisms/Heatmap/index.tsx

```diff
@@ -38,12 +47,54 @@ export function Heatmap({
   const headers = generateHeaders(areas, groupAreaCode);
   const rows = generateRows(areas, indicators, dataPoints);
 
+  const idPrefix = 'hover'; //useId(); // - useId breaks Next hydration for some reason
+  const getHoverId = (cellKey: string) => {
+    return `${idPrefix}-${cellKey}`;
+  };
+
+  const handleMouseOverCell = (event: React.MouseEvent, hoverId: string) => {
+    const hoverElement = document.getElementById(hoverId);
+    if (!hoverElement) {
+      return;
+    }
+    const cellRect = event.currentTarget.getBoundingClientRect();
+
+    hoverElement.style.display = 'block';
+    hoverElement.style.left = `${cellRect?.right + 12}px`;
+    hoverElement.style.top = `${cellRect?.top + (cellRect?.bottom - cellRect?.top) / 2}px`;
+  };
+
+  const handleMouseLeaveCell = (hoverId: string) => {
+    const hoverElement = document.getElementById(hoverId);
+    if (!hoverElement) {
+      return;
+    }
+
+    hoverElement.style.display = 'none';
+  };
+
   return (
     <>
       <BenchmarkLegend />
+      {rows.flatMap((row) => {
+        return row.cells.map((cell) => {
+          return cell.hoverProps ? (
+            <HeatmapHover
+              key={`hover-${cell.key}`}
+              areaName={cell.hoverProps.areaName}
+              period={cell.hoverProps.period}
+              indicatorName={cell.hoverProps.indicatorName}
+              value={cell.hoverProps.value}
+              unitLabel={cell.hoverProps.unitLabel}
+              benchmark={cell.hoverProps.benchmark}
+              hoverId={getHoverId(cell.key)}
+            />
+          ) : null;
+        });
+      })}
       <StyledDivTableContainer>
         <StyledTable data-testid="heatmapChart-component">
-          <Table.Row>
+          <StyledRow>
             {headers.map((header) => {
               return (
                 <HeatmapHeader
```

Comment:
```suggestion
            {headers.map((header) => (
                <HeatmapHeader
```
If you use normal ( instead of curly ones you can omit the return, makes it a little more concise